lib/_tls_wrap.js

```diff
@@ -1138,8 +1140,17 @@ exports.connect = function(...args /* [port,] [host,] [options,] [cb] */) {
   if (options.session)
     socket.setSession(options.session);
 
-  if (options.servername)
+  if (options.servername) {
+    if (ipServernameWarned === false && net.isIP(options.servername)) {
```

Comment:
Using `!ipServernameWarned` instead of `ipServernameWarned === false` might be better.

Reply:
Just a quick note here that may be worth noting in the docs... the `net.isIP()` function returns 0 if an ipv6 address is given with the surrounding `[` `]` ... e.g. `net.isIP('[fe80::a00:27ff:fec0:eb1a]') === 0`. This is, of course, correct because the square brackets are not actually a part of the address, but it could catch some users.

Reply:
I went with `=== false` to match an existing one-time warning (dunno if it optimizes slightly better or what); I can switch to `!` if you prefer.

`url.parse('https://[fe80::a00:27ff:fec0:eb1a]:3333/')`, for instance, fills `hostname` with `fe80::a00:27ff:fec0:eb1a` (bracketless), and if you manually build a URL with brackets around an IPv6 address in `hostname`, `https.get()` on it fails on DNS lookup, so I _think_ we should be fine there?